### PR TITLE
Structure + Entities (in progress) added.

### PR DESCRIPTION
**Pull request by kmavropulo**
Structure + entities (now the in refactoring stage). Plan of task implementation is in Readme.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,16 +1,31 @@
...
+    testCompile 'org.junit.jupiter:junit-jupiter-api:+',
+            'org.hamcrest:java-hamcrest:+'
+
+    compile 'ch.qos.logback:logback-classic:+',
+            'com.h2database:h2:+',
+            'org.glassfish.web:javax.servlet.jsp.jstl:+',
+            'org.glassfish:javax.el:3.0.1-b08'
+
+    compile group: 'org.hibernate', name: 'hibernate-core', version: '5.2.12.Final'
+    compile group: 'org.hibernate.validator', name: 'hibernate-validator', version: '6.0.4.Final'
```
Comment (SMishAlex) [inline build.gradle:30]: please use same style for all dependencies

File: src/main/java/com/github/admissionCommittee/Main.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+/**
+ * Created by Admin on 07.11.2017.
+ */
+public class Main {
+    public static void main(String[] args) {
+        SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+        Session session = sessionFactory.openSession();
+        session.beginTransaction();
+        Role role = new Role("admin");
```
Comment (SMishAlex) [inline src/main/java/com/github/admissionCommittee/Main.java]: bad naming, role and role1 looks awful

File: src/main/java/com/github/admissionCommittee/Main.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        //transaction.rollback();
+        transaction.commit();
+
+        session.beginTransaction();
+        //User load = session.load(User.class, 0);
+        //Role loadedRole = session.load(Role.class, 0);
+        //System.out.println(user.getName());
+        //System.out.println(loadedRole.getName());
+
+        List userResult = session.createQuery("from User").list();
```
Comment (SMishAlex) [inline src/main/java/com/github/admissionCommittee/Main.java]: don't use raw type, use <?> at least

File: src/main/java/com/github/admissionCommittee/model/BaseEntity.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+import javax.persistence.*;
+
+@MappedSuperclass
+public class BaseEntity implements HasId {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "id", unique = true, nullable = false)
+
+    private Integer id;
```
Comment (SMishAlex) [inline src/main/java/com/github/admissionCommittee/model/BaseEntity.java:14]: don't separate fields with their notations

File: src/main/java/com/github/admissionCommittee/model/BaseEntity.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+    protected BaseEntity() {
+    }
+
+    protected BaseEntity(Integer id) {
+        this.id = id;
+    }
```
Comment (SMishAlex) [inline src/main/java/com/github/admissionCommittee/model/BaseEntity.java:21]: lombok can do it for you but I don't insist

File: src/main/java/com/github/admissionCommittee/model/Faculty.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+@Table(name = "faculties")
+@AttributeOverride(name = "id", column = @Column(name = "faculties_id",
+        nullable = false))
+public class Faculty extends NamedEntity {
+
+    @Column(name = "max_students")
+    private int maxStudents;
+
+    // TODO: map to RequiredSubject instance using Hibernate annotations
+    private Map<String, Object> requiredSubjects;
```
Comment (SMishAlex) [inline src/main/java/com/github/admissionCommittee/model/Faculty.java:21]: why do we use Object as a value?

File: src/main/java/com/github/admissionCommittee/model/User.java [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+    @Column(name = "passport")
+    private String passport;
+    @Column(name = "email")
+    private String email;
+    @Column(name = "phone")
+    private String phone;
+    @OneToMany(mappedBy = "user")
+    @Column(name = "faculty")
+    private Set<Faculty> faculty;
+    @OneToOne(fetch = FetchType.EAGER)
```
Comment (SMishAlex) [inline src/main/java/com/github/admissionCommittee/model/User.java:31]: EAGER is actually bad... it's better to use LAZY + Transactional when you need it

File: src/main/java/com/github/admissionCommittee/util/HibernateUtil.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+import com.github.admissionCommittee.model.User;
+import org.hibernate.SessionFactory;
+import org.hibernate.boot.registry.StandardServiceRegistryBuilder;
+import org.hibernate.cfg.Configuration;
+
+/**
+ * Created by Admin on 07.11.2017.
+ */
+public class HibernateUtil {
+    private static SessionFactory sessionFactory = null;
```
Comment (SMishAlex) [inline src/main/java/com/github/admissionCommittee/util/HibernateUtil.java]: it will null by default

**Comment by kmavropulo** [comment]
DAO, models and Services is under construction, we shall fix the mistakes and create new pull request as soon as possible. Thank you for the review.

**Pull request closed without merging**